crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Repository
+    attr_reader :platform, :id, :registry, :available, :md5, :url
+
+    class << self
+      #
+      # class methods
+      #
+      def load_registry
```

Comment:
`require_registry` and `load!` both correctly reflect that it might or might not load it, but they also both make it sound like it's purely an action method which does stuff but doesn't return anything.  In this case the caller cares more about the return value than how it's implemented behind the scenes - the caller doesn't need to know whether it's memoized or not.  So IMHO I still think `registry` is the best name.